### PR TITLE
Feat/ingredient module

### PR DESCRIPTION
**Pull request by RanNissan**

### CODE REVIEW

File: Controllers/IngredientController.cs [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+        // GET: Ingredient
+        public async Task<ActionResult> Index()
+        {
+
+            return View(this.GetIngredients());
+        }
+
+        public List<Ingredient> GetIngredients()
+        {
+            return db.Ingredients.ToList<Ingredient>().ToList<Ingredient>();
```
Comment (shaySirek) [inline Controllers/IngredientController.cs:28]: why are there two ToList calls?

File: Controllers/IngredientController.cs [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+            {
+                return Redirect("/Error/Unauthorized");
+            }*/
+            return View(ingredient);
+        }
+
+        /*// POST: Ingredient/Delete/5
+        [HttpPost, ActionName("Delete")]
+        [ValidateAntiForgeryToken]
+        public async Task<ActionResult> DeleteConfirmed(int id)
```
Comment (omerzamir) [inline Controllers/IngredientController.cs:135]: Is it commented because recipe entity it's missing?

**Comment by shaySirek** [comment]
[Ingredient](https://trello.com/c/1JIPqVDv/25-ingredient)

File: Controllers/IngredientController.cs (hunk 1 above)
Comment (RanNissan) [reply Controllers/IngredientController.cs]: Gonna fix..

File: Controllers/IngredientController.cs (hunk 2 above)
Comment (RanNissan) [reply Controllers/IngredientController.cs:131]: Yep.. I havnt touched this method yet..

File: Controllers/IngredientController.cs [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+            if (id == null)
+            {
+                return new HttpStatusCodeResult(HttpStatusCode.BadRequest);
+            }
+
+            Ingredient ingredient = await bl.GetIngredientById(id);
+
+            if (ingredient == null)
+            {
+                return Redirect("/Error/NotFound");
```
Comment (shaySirek) [inline Controllers/IngredientController.cs:38]: there is no /Error/NotFound view.
could you create it?

File: Controllers/IngredientController.cs [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+
+            return View(bl.GetIngredients());
+        }
+
+        // GET: Ingredient/Details/5
+        public async Task<ActionResult> Details(int? id)
+        {
+            if (id == null)
+            {
+                return new HttpStatusCodeResult(HttpStatusCode.BadRequest);
```
Comment (shaySirek) [inline Controllers/IngredientController.cs:31]: maybe we should make /Error/BadRequest view for the user experience.

File: Controllers/IngredientController.cs (hunk 3 above)
Comment (RanNissan) [reply Controllers/IngredientController.cs:38]: It should be in another task..

File: Controllers/IngredientController.cs (hunk 4 above)
Comment (omerzamir) [reply Controllers/IngredientController.cs:31]: https://dzone.com/articles/make-custom-error-pages-work-in-aspnet-mvc-5

Comment (RanNissan) [reply Controllers/IngredientController.cs:31]: Not critical..

**Pull request merged**